### ISSUE TITLE
Add logic to delete Solr cores as part of instance and bulk delete

### ISSUE THREAD

**Issue by dsolovay** [dsolovay/Sitecore-Instance-Manager#40]
Currently, Solr cores are left in place if a site is deleted. Functionality can be added to fix this by adding logic to call

    http://localhost:8983/solr/admin/cores?action=UNLOAD&deleteIndex=true&deleteInstanceDir=true&core=CORE_NAME

The Solr URL is in the Solr root config, and the name of each core can be pulled from the individual index configurations. The DELETE information screen text should also be updated.

**Comment by AlenPelin** [comment]
We should be very careful implementing this feature because there could be several instances sharing same Solr Cores. My suggestion is to create popup that asks user for confirmation before proceeding with delete command.

**Comment by dsolovay** [comment]
What if we added a configuration option, Core/Delete/SolrIndexes, and set the default to "False"?  This would simplify things for users that used the SIM script to create a number of instances, and would work better with automated scripts.
![image](https://cloud.githubusercontent.com/assets/689532/20317591/af79d520-ab34-11e6-9521-d6b8d74fd4b1.png)

**Comment by dsolovay** [comment]
@AlenPelin What if we only bulk deleted the index if the name of the instance was embedded? So an instance called sc82u1 could delete its indexes if they all started with sc82u1 (e.g. sc82u1_sitecore_master_index, etc.) but could not delete an index called `sitecore_master_index`. This would make the risk of deleting a shared index extremely slight, and would greatly help a developer who has created a large number of instances and needs to clean up.

This behavior would delete indexes that were created by the plugin, but would not delete indexes that were created by manually enabling the config files.

**Comment by AlenPelin** [comment]
I think we can ask for confirmation before deleting each core the same way as SIM asks before deleting SQL database that is stored out of instance's root folder. 
The reason is that Solr cores can be shared between several instances which is not trivial to check.

**Comment by dsolovay** [comment]
> The reason is that Solr cores can be shared between several instances which is not trivial to check.

Doesn't checking that the index begins with the instance name solve this problem? 

At 15 indexes per instance, prompting on each one will become very tedious, especially if you are cleaning up multiple instances. I had >100 cores at one point.

**Comment by AlenPelin** [comment]
Agree, it makes sense to aggregate indexes by similar prefix and show popups with batches.
